### ISSUE TITLE
Any way to release android microfone when mute

### ISSUE THREAD

**Issue by EugeneGxUA** [livekit/client-sdk-android#375]
Is there is any wayt to release micro of device when mute your self ?

Now if we want to release micro we need unpublishAudioTrack but this create history overflow at server side, when all users publish and unpublish tracks.

Maybe somehow we can setMicroEnabled(FALSE) and release device micro ?

Thanks

**Comment by adrian-niculescu** [comment]
There are two ways to do this today and neither is quite right.

**1. Unpublish and republish the audio track.** Releases the mic, but leaks native memory on the audio path: 50 mic toggles on a two-party call retained about 11.7 MB of native heap, roughly 0.23 MB per cycle, with no decay afterwards. The video path got a fix for the same shape of leak in #971, and audio has no counterpart to it. And as the original post says, publish/unpublish is not free on the server either.

**2. Mute the publication.** `LocalTrackPublication.muted` only sets `mediaTrack.enabled = false`. It never touches the ADM, so `AudioRecord` stays open and the OS microphone-in-use indicator stays lit. An indicator burning next to a muted mic reads as a bug at best and a privacy invasion at worst.

The web SDK handles this with [`stopMicTrackOnMute`](https://github.com/livekit/client-sdk-js/blob/41c19491aeda7c3e1be16d04a8086d2e4276f291/src/room/track/options.ts#L132), which stops the underlying `MediaStreamTrack` so the browser releases the device. That has no direct port to Android, since capture is owned by the ADM shared across the `PeerConnectionFactory` rather than by the track. `PeerConnection.setAudioRecording(false)` is the equivalent lever: it stops the ADM and releases `AudioRecord` with no renegotiation and no change to the publication.

#987 adds that as `LocalAudioTrackOptions.stopMicrophoneOnMute`.

@vhimanshu07 `AudioManager.isMicrophoneMute` only zeroes the samples, it does not release the device. The assertion you hit is in the prewarm path that `setMicrophoneEnabled` drives on every toggle, which is the same churn option 1 causes.

**Comment by davidliu** [comment]
@adrian-niculescu hmm, I'm not able to repro this. By default, libwebrtc should be releasing the audio record when muted and subsequently turning off the mic indicator. [There's actually a PR to allow for the opposite](https://github.com/webrtc-sdk/webrtc/pull/269). Do you have a repro example?

**Comment by adrian-niculescu** [comment]
Apologies @davidliu, you're right. I went to build a repro and couldn't reproduce it either: on an unmodified v2.28.0, muting the publication releases AudioRecord and the mic indicator goes off.

I've been carrying the unpublish/republish workaround since long before v2.23.1 and didn't realize the WebRTC bump in that release (https://github.com/webrtc-sdk/webrtc/pull/206, stop recording when all send streams are muted) had essentially fixed this. When I recently decided to help fix it properly so that mute/unmute releases the mic, the web SDK still having `stopMicTrackOnMute` reinforced my assumption that nothing had changed, so I went straight to implementing without retesting the default path first.

Closing https://github.com/livekit/client-sdk-android/pull/987. For anyone landing on this issue: since v2.23.1, muting your audio publication releases the microphone by default, no unpublish needed.

**Comment by davidliu** [comment]
Yeah, I think we had a regression in libwebrtc around that time where the default stop on mute behavior stopped working. Closing this issue.
